returnn/oggzip.py

```diff
@@ -1,104 +1,200 @@
+    def run(self, task_id):
+        output = (
+            self.zip_subarchives.hidden_paths[task_id]
+            if self.concurrent > 1
+            else "out.ogg.zip"
+        )
+        args = [
+            tk.uncached_path(self.returnn_python_exe),
+            os.path.join(
+                tk.uncached_path(self.returnn_root), "tools/bliss-to-ogg-zip.py"
+            ),
+            tk.uncached_path(self.bliss_corpus),
+            "--output",
+            output,
+        ]
+        if isinstance(self.segments, MultiOutputPath):
+            args.extend(["--subset_segment_file", self.segments.hidden_paths[task_id]])
+        elif self.segments is not None:
+            args.extend(["--subset_segment_file", tk.uncached_path(self.segments)])
+
+        if self.no_audio:
+            args.extend(["--no_ogg"])
+        elif self.no_conversion:
+            args.extend(["--no_conversion"])
+        else:
+            if self.rasr_cache is not None:
+                args.extend(["--sprint_cache", tk.uncached_path(self.rasr_cache)])
+            if self.raw_sample_rate is not None:
+                args.extend(["--raw_sample_rate", str(self.raw_sample_rate)])
+            if self.feat_sample_rate is not None:
+                args.extend(["--feat_sample_rate", str(self.feat_sample_rate)])
+
+        sp.check_call(args)
+        if self.concurrent == 1:
+            relink("out.ogg.zip", self.out_ogg_zip.get_path())
+
+    def merge(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            # extract all subarchives
+            for zip_subarchive in self.zip_subarchives.hidden_paths.values():
+                with zipfile.ZipFile(
+                    zip_subarchive, mode="r", compression=zipfile.ZIP_DEFLATED
+                ) as zip_file:
+                    zip_file.extractall(tmp_dir)
+                os.remove(zip_subarchive)
+
+            # create output folder
+            assert self.out_ogg_zip.get().endswith(".zip")
+            output_folder = os.path.join(
+                tmp_dir, os.path.basename(self.out_ogg_zip.get())[: -len(".zip")]
+            )
+            os.mkdir(output_folder)
+
+            # merge meta files, remove from subarchives
+            meta_file = output_folder + ".txt"
+            with open(meta_file, "w") as mf:
+                mf.write("[\n")
+                for zip_subarchive in self.zip_subarchives.hidden_paths.values():
+                    sub_meta_file = os.path.join(
+                        tmp_dir, zip_subarchive.replace(".zip", ".txt")
+                    )
+                    with open(sub_meta_file, "r") as smf:
+                        for line in smf.readlines():
+                            if not (line.startswith("[") or line.startswith("]")):
+                                mf.write(line)
+                    os.remove(sub_meta_file)
+                mf.write("]\n")
+
+            # move all folders to output folder
+            cmd = [
+                "rsync",

```

Comment:
I am wondering. `rsync` is a default ubuntu package, if I remember correctly. Is there any concern about other operating systems?

Reply:
Do we care?

Reply:
I would argue we don't as the job will use other packages (`ffmpeg` in the called RETURNN script, possibly more) as well which might not be supported on other systems either.